### ISSUE TITLE
OCaml build of F* generates invalid constructors when extracting to F#

### ISSUE THREAD

**Issue by A-Manning** [FStarLang/FStar#1242]
The OCaml build is adding bogus `U__` prefixes to constructor identifiers when extracting F#. 
The F# build seems to be ok here. 

```ocaml
val maybe(#a #b:Type): b -> (a -> b) -> option a -> b
let maybe #_ #_ y f = function | None -> y
                               | Some x -> f x
```
Extracts to F# as
```fsharp
let maybe = (fun ( y  :  'Auu___5_19 ) ( f  :  'Auu___4_18  ->  'Auu___5_19 ) ( uu___6_36  :  'Auu___4_18 FStar.Pervasives.Native.option ) -> (match (uu___6_36) with
| FStar.Pervasives.Native.U__None -> begin
y
end
| FStar.Pervasives.Native.U__Some (x) -> begin
(f x)
end))
```

But extracts to OCaml as
```ocaml
let maybe:
  'Auu___4_18 'Auu___5_19 .
    'Auu___5_19 ->
      ('Auu___4_18 -> 'Auu___5_19) ->
        'Auu___4_18 FStar_Pervasives_Native.option -> 'Auu___5_19
  =
  fun y  ->
    fun f  ->
      fun uu___6_36  ->
        match uu___6_36 with
        | FStar_Pervasives_Native.None  -> y
        | FStar_Pervasives_Native.Some x -> f x
```

Note the `FStar.Pervasives.Native.U__None` vs. the correct `FStar_Pervasives_Native.None`

**Comment by iamrecursion** [comment]
Just to clarify, do you mean “when extracting F*” at the start?

**Comment by A-Manning** [comment]
@iamrecursion I mean "when using the OCaml-generated F* binary".

**Comment by iamrecursion** [comment]
Thanks for the clarification.

**Comment by A-Manning** [comment]
Fixed in #1192
